src/main/java/com/google/api/gax/bundling/BundlingThreshold.java

```diff
@@ -34,22 +34,28 @@
 /**
  * The interface representing a threshold to be used in ThresholdBundler.
  * Thresholds do not need to be thread-safe if they are only used inside
- * of ThresholdBundler.
+ * ThresholdBundler.
  */
 public interface BundlingThreshold<E> {
 
   /**
    * Presents the element to the threshold for the attribute of interest to be accumulated.
+   *
+   * Any calls into this function from ThresholdBundler will be under a lock.
    */
   void accumulate(E e);
 
   /**
+   * Any calls into this function from ThresholdBundler will be under a lock.
+   *
    * @return whether the threshold has been reached.
    */
   boolean isThresholdReached();
 
   /**
-   * Reset the accumulated value back to zero.
+   * Make a copy of this threshold with the accumulated value reset.
+   *
+   * Any calls into this function from ThresholdBundler will be under a lock.
    */
-  void reset();
+  BundlingThreshold<E> copyReset();
```

Comment:
Yeah, I agree that the name is unclear. It is unfortunate that 'reset' doesn't have a distinct past tense form. 

'copyAndReset' doesn't quite work because it seems to indicate that the threshold that this method is called on will get reset, but it's actually the one being returned that has a "reset" threshold. Maybe "copyWithZeroedValue"?